Description: Add --python-import-prefix CLI option to flatc compiler.
    This supports import redirection under complex monorepo layouts,
    preventing import resolution errors when generated Python
    submodules are loaded within large namespace packages.
    Implemented by declaring python_import_prefix in IDLOptions
    (include/flatbuffers/idl.h), adding command-line argument parsing
    for --python-import-prefix in src/flatc.cpp, and modifying the
    Python generator in src/idl_gen_python.cpp to prepend the prefix
    followed by a dot (e.g., from <prefix>.<module> import) on all
    generated Python dependency import strings (excluding typing and
    flatbuffers built-in library modules).


diff --git a/include/flatbuffers/idl.h b/include/flatbuffers/idl.h
index 3e35b48..35e4eae 100644
--- a/include/flatbuffers/idl.h
+++ b/include/flatbuffers/idl.h
@@ -733,6 +733,10 @@ struct IDLOptions {
   // Whether to generate numpy helpers.
   bool python_gen_numpy;
 
+  // If specified, this is added as a prefix to the paths of generated modules
+  // in import statements.
+  std::string python_import_prefix;
+
   bool ts_omit_entrypoint;
   ProtoIdGapAction proto_id_gap_action;
 
diff --git a/src/flatc.cpp b/src/flatc.cpp
index 4bb590b..f5c0cb3 100644
--- a/src/flatc.cpp
+++ b/src/flatc.cpp
@@ -708,6 +708,17 @@ FlatCOptions FlatCompiler::ParseFromCommandLineArguments(int argc,
       } else if (arg == "--no-python-gen-numpy" ||
                  arg == "--python-gen-numpy=false") {
         opts.python_gen_numpy = false;
+      } else if (arg.rfind("--python-import-prefix=", 0) == 0) {
+        opts.python_import_prefix =
+            arg.substr(std::string("--python-import-prefix=").size());
+      } else if (arg == "--python-import-prefix") {
+        if (++argi >= argc) {
+          Error("Missing value following: " + arg, true);
+        }
+        if (argv[argi][0] == '-') {
+          Error("Invalid value for " + arg + ": " + argv[argi], true);
+        }
+        opts.python_import_prefix = argv[argi];
       } else if (arg == "--ts-omit-entrypoint") {
         opts.ts_omit_entrypoint = true;
       } else if (arg == "--annotate-sparse-vectors") {
diff --git a/src/idl_gen_python.cpp b/src/idl_gen_python.cpp
index d325fee..b8e70bc 100644
--- a/src/idl_gen_python.cpp
+++ b/src/idl_gen_python.cpp
@@ -2853,13 +2853,21 @@ class PythonGenerator : public BaseGenerator {
         code += "from typing import Any\n";
 
         for (auto import_entry : imports) {
+          std::string prefix{""};
+          // TODO(Sanjay): Can we do better with detecting import entries for
+          // generated modules here?
+          if (!parser_.opts.python_import_prefix.empty() &&
+              import_entry.first != "typing" &&
+              import_entry.first != "flatbuffers.table") {
+            prefix = parser_.opts.python_import_prefix + ".";
+          }
           // If we have a file called, say, "MyType.py" and in it we have a
           // class "MyType", we can generate imports -- usually when we
           // have a type that contains arrays of itself -- of the type
           // "from .MyType import MyType", which Python can't resolve. So
           // if we are trying to import ourself, we skip.
           if (import_entry.first != local_import) {
-            code += "from " + import_entry.first + " import " +
+            code += "from " + prefix + import_entry.first + " import " +
                     import_entry.second + "\n";
           }
         }
